time/src/test/java/net/jqwik/time/api/TimesConstraintTests.java

```diff
@@ -0,0 +1,1104 @@
+package net.jqwik.time.api;
+
+import java.time.*;
+import java.time.format.*;
+import java.util.*;
+
+import net.jqwik.api.*;
+import net.jqwik.testing.*;
+import net.jqwik.time.api.constraints.*;
+
+import static java.time.temporal.ChronoUnit.*;
+import static org.assertj.core.api.Assertions.*;
+
+import static net.jqwik.api.Arbitraries.*;
+
+@Group
+public class TimesConstraintTests {
+
+	@Group
+	class LocalTimeConstraints {
+
+		@Group
+		class Constraints {
+
+			@Property
```

Comment:
There's a test missing (in other places as well) along the following lines:
```java
@Property
void timeRangeForProvidedTime(@ForAll("myTimes") @TimeRange(min = "00:01:00", max = "00:02:00") LocalTime time) {
	assertThat(time).isBetween(
			LocalTime.of(0, 1, 0),
			LocalTime.of(0, 2, 0)
	);
}

@Provide
Arbitrary<LocalTime> myTimes() {
	return Arbitraries.of(
			LocalTime.of(0, 0, 1),
			LocalTime.of(0, 1, 1),
			LocalTime.of(0, 2, 1)
	);
}
```

The point is to allow the constraints to work even when a self-made arbitrary generates the time (or whatever). 

Implementationwise this can be done by a simple filter in those cases. And then you probably want to split `TimeRangeConfigurator` in two classes: `TimeRangeForLocalTimeConfigurator` and `TimeRangeForOffsetTimeConfigurator` in order to minimize the number of `instanceof` usages.

And this holds for most other time-related constraint annotations as well I'm afraid.

Reply:
How should I handle Periods? 

The problem with periods: For example a Period can have 3 Years, 293 Months and 211 days. How can I compare it with a period with for example 4 Years, 2 Month and 11 days?

Reply:
I don't see the relation to my comment above, but anyway: you could compare by effective length. Or maybe a period range does not make sense.

Reply:
My comment is here because I'll change all configurators so that they work for user arbitraries.

Reply:
Done.